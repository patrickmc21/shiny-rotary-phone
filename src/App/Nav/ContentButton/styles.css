.content-button {
  height: 80px;
  width: 160px;
  font-size: 28px;
}