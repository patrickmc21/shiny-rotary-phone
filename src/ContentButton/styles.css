.content-button {
  border: 1px solid #66AFCF;
  border-radius: 30px;
  height: 70px;
  width: 12vw;
  background-color: transparent;
  box-shadow: 0px 0px 5px 1px #66AFCF;
  background-size: cover;
  background-repeat: no-repeat;
  background-size: 100% 100%;
  background-position: 0 2px;
}

.content-button.active {
  border: 1px solid #F3BF44;
  box-shadow: 0px 0px 5px 1px #F3BF44;
}

.content-button:hover {
  border: 1px solid #9F1F16;
  box-shadow: 0px 0px 5px 1px #9F1F16;
}

.content-button.people {
  background-image: url('./images/people.png');
}

.content-button.vehicles {
  background-image: url('./images/vehicles.png');
}

.content-button.starships {
  background-image: url('./images/starships.png');
}

.content-button.planets {
  background-image: url('./images/planets.png');
}

.content-button.people.active {
  background-image: url('./images/people-active.png');
}

.content-button.vehicles.active {
  background-image: url('./images/vehicles-active.png');
}

.content-button.starships.active {
  background-image: url('./images/starships-active.png');
}

.content-button.planets.active {
  background-image: url('./images/planets-active.png');
}

.content-button.people:hover {
  background-image: url('./images/people-hover.png');
}

.content-button.vehicles:hover {
  background-image: url('./images/vehicles-hover.png');
}

.content-button.starships:hover {
  background-image: url('./images/starships-hover.png');
}

.content-button.planets:hover {
  background-image: url('./images/planets-hover.png');
}
