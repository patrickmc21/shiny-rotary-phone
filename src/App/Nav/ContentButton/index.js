import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentButton = () => {
  return(
    <div>I'm a content button</div>
  )
}

export default ContentButton;