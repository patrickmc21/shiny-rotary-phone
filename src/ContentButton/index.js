import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const ContentButton = ({buttonType, activateCategory, activeCategoryName}) => {
  const className = buttonType === activeCategoryName ? 
    `content-button ${buttonType} active` 
    : `content-button ${buttonType}`;
  return (
    <button 
      className={className}
      onClick={() => activateCategory(buttonType)}>
    </button>
  );
};

ContentButton.propTypes = {
  buttonType: PropTypes.string,
  activateCategory: PropTypes.func,
  activeCategoryName: PropTypes.string
};

export default ContentButton;