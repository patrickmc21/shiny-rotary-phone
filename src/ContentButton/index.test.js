import React from 'react';
import { shallow, mount } from 'enzyme';
import ContentButton from './index.js';

describe('ContentButton', () => {
  let wrapper;
  let mockButtonType;
  let mockActivateCategory;

  beforeEach( () => {
    mockButtonType = 'people';
    mockActivateCategory = jest.fn();
    wrapper = shallow(
      <ContentButton 
        buttonType={mockButtonType}
        activateCategory={mockActivateCategory}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should run activateCategory with correct params on click', () =>{
    const expected = mockButtonType;
    wrapper.find('.content-button').simulate('click');
    expect(mockActivateCategory).toHaveBeenCalledWith(expected)
  })
})