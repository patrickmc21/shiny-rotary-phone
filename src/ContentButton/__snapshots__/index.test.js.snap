// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentButton add an active class if current category is the button type 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentButton
    activateCategory={[Function]}
    activeCategoryName="people"
    buttonType="people"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "content-button people active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "content-button people active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ContentButton should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentButton
    activateCategory={[Function]}
    activateCategoryName="vehicles"
    buttonType="people"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "content-button people",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "content-button people",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
